config/config.go

```diff
@@ -419,7 +444,10 @@ func loadExporters(v *viper.Viper, factories map[configmodels.Type]component.Exp
 
 func loadProcessors(v *viper.Viper, factories map[configmodels.Type]component.ProcessorFactory) (configmodels.Processors, error) {
 	// Get the list of all "processors" sub vipers from config source.
-	processorsConfig := ViperSub(v, processorsKeyName)
+	processorsConfig, err := ViperSubExact(v, processorsKeyName)
+	if err != nil {
+		return nil, err
```

Comment:
This cannot happen and impossible to test because of https://github.com/open-telemetry/opentelemetry-collector/pull/1641/files#diff-3baf47c64847a8fb8aaa8cc2e088513bR116, where this is guaranteed to be a map. If it is not will fail sooner at line https://github.com/open-telemetry/opentelemetry-collector/pull/1641/files#diff-3baf47c64847a8fb8aaa8cc2e088513bR124